include/small/ibuf.h

```diff
@@ -196,6 +202,24 @@ ibuf_truncate(struct ibuf *ibuf, size_t used)
 	ibuf_discard(ibuf, ibuf_used(ibuf) - used);
 }
 
+/** Consume size bytes of data from the read end of the buffer. */
+static inline void
+ibuf_consume(struct ibuf *ibuf, size_t size)
+{
+	assert(size <= ibuf_used(ibuf));
+	ASAN_POISON_MEMORY_REGION(ibuf->rpos, size);
```

Comment:
Let's add `ibuf_poison_consumed` helper for the sake of symmetry.

Reply:
Fixed. Also added tests for poisoning.

<details>
<summary>diff</summary>

```diff
diff --git a/include/small/ibuf.h b/include/small/ibuf.h
index 0c5d89e..5cb6b9e 100644
--- a/include/small/ibuf.h
+++ b/include/small/ibuf.h
@@ -125,6 +125,12 @@ ibuf_unpoison_unallocated(const struct ibuf *ibuf)
 	ASAN_UNPOISON_MEMORY_REGION(ibuf->wpos, ibuf->end - ibuf->wpos);
 }
 
+static inline void
+ibuf_poison_consumed(const struct ibuf *ibuf)
+{
+	ASAN_POISON_MEMORY_REGION(ibuf->buf, ibuf->rpos - ibuf->buf);
+}
+
 static inline void
 ibuf_unpoison_consumed(const struct ibuf *ibuf)
 {
@@ -207,8 +213,8 @@ static inline void
 ibuf_consume(struct ibuf *ibuf, size_t size)
 {
 	assert(size <= ibuf_used(ibuf));
-	ASAN_POISON_MEMORY_REGION(ibuf->rpos, size);
 	ibuf->rpos += size;
+	ibuf_poison_consumed(ibuf);
 }
 
 /** Consume bytes before ptr from the read end of the buffer. */
diff --git a/test/ibuf.c b/test/ibuf.c
index 7b66f4a..9665365 100644
--- a/test/ibuf.c
+++ b/test/ibuf.c
@@ -297,6 +297,39 @@ test_ibuf_poison(void)
 	for (char *p = ibuf.wpos; p < ibuf.end; p++)
 		fail_unless(__asan_address_is_poisoned(p));
 
+	/* Test poison on consume. */
+	ibuf_consume(&ibuf, 100);
+	memset(ibuf.rpos, 0, ibuf_used(&ibuf));
+	char *poison_end;
+	poison_end = (char *)small_align_down((uintptr_t)ibuf.rpos,
+					      SMALL_POISON_ALIGNMENT);
+	for (char *p = ibuf.buf; p < poison_end; p++)
+		fail_unless(__asan_address_is_poisoned(p));
+
+	/* Test no unpoisoned area left between this and previous consume. */
+	ibuf_consume(&ibuf, 77);
+	memset(ibuf.rpos, 0, ibuf_used(&ibuf));
+	poison_end = (char *)small_align_down((uintptr_t)ibuf.rpos,
+					      SMALL_POISON_ALIGNMENT);
+	for (char *p = ibuf.buf; p < poison_end; p++)
+		fail_unless(__asan_address_is_poisoned(p));
+
+	/* Test poison on consume_before. */
+	ibuf_consume(&ibuf, 150);
+	memset(ibuf.rpos, 0, ibuf_used(&ibuf));
+	poison_end = (char *)small_align_down((uintptr_t)ibuf.rpos,
+					      SMALL_POISON_ALIGNMENT);
+	for (char *p = ibuf.buf; p < poison_end; p++)
+		fail_unless(__asan_address_is_poisoned(p));
+
+	/* Test no unpoisoned area left between this and previous consume. */
+	ibuf_consume(&ibuf, 44);
+	memset(ibuf.rpos, 0, ibuf_used(&ibuf));
+	poison_end = (char *)small_align_down((uintptr_t)ibuf.rpos,
+					      SMALL_POISON_ALIGNMENT);
+	for (char *p = ibuf.buf; p < poison_end; p++)
+		fail_unless(__asan_address_is_poisoned(p));
+
 	/* Test poison on shrink. */
 	ibuf_reset(&ibuf);
 	ptr = ibuf_alloc(&ibuf, 333);
```

</details>